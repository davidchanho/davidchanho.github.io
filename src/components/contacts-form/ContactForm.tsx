import Section from 'components/section/Section';
import emailjs from 'emailjs-com';
import { useLocalStorage } from 'hooks/useLocalStorage';
import React, { ChangeEvent, FormEvent } from "react";
import { Button, Form } from "react-bootstrap";
import { serviceId, templateId, userId } from '../../config.json';

interface ContactFormProps {
  name: string;
  email: string;
  message: string;
}

const initialForm: ContactFormProps = { name: '', email: '', message: '' }

function ContactForm() {
  const [stored, setStored] = useLocalStorage('form', `${initialForm}`);

  const handleSubmit = (e: FormEvent<HTMLFormElement | HTMLButtonElement>) => {
    e.preventDefault();

    emailjs.sendForm(serviceId, templateId, '#contactForm', userId)
      .then(function (response) {
        console.log('SUCCESS!', response.status, response.text);
        setStored(initialForm);
      }, function (error) {
        console.log('Error');
      });

    setStored(initialForm)
  };

  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
    const { name, value } = e.target;

    setStored({
      ...stored,
      [name]: value
    })
  }

  const handleClear = () => {
    localStorage.removeItem('form');
  }

  return (
    <Section name='contactForm' header='Contact Me'>
      <Form id='contactForm' className='w-50 vh-100 my-5 d-flex flex-column mx-auto ' onSubmit={handleSubmit}>
        <Form.Group controlId="formName">
          <Form.Label>Name</Form.Label>
          <Form.Control defaultValue={stored.name} onChange={handleChange} name="name" type="text" placeholder="Enter name" />
        </Form.Group>

        <Form.Group controlId="formEmail">
          <Form.Label>Email address</Form.Label>
          <Form.Control onChange={handleChange}
            name="email"
            type="email"
            placeholder="Enter email"
            defaultValue={stored.email}
          />
        </Form.Group>

        <Form.Group controlId="formMessage">
          <Form.Label>Message</Form.Label>
          <Form.Control onChange={handleChange}
            name="message"
            type="text"
            placeholder="Enter message"
            as="textarea"
            rows={10}
            defaultValue={stored.message}
            style={{ resize: 'none' }}
          />
        </Form.Group>

        <div className='d-flex align-items-center justify-content-end'>
          <Button className="mr-2" variant="danger" onSubmit={handleClear}>
            Clear
        </Button>
          <Button type="submit" variant="success" onSubmit={handleSubmit}>
            Submit
        </Button>
        </div>
      </Form>
    </Section>
  );
}

export default ContactForm;
