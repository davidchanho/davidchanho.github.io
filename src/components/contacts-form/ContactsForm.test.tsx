import React from "react";
import renderer from "react-test-renderer";
import ContactForm from ".";

describe("ContactForm", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<ContactForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
