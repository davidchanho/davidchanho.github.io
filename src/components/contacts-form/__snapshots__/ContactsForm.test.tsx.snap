// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm renders correctly 1`] = `
<div
  className="sc-bdfBwQ fKNFrQ"
  name="contactForm"
>
  <h3
    className="text-center text-capitalize mb-5"
  >
    <u>
      Contact Me
    </u>
  </h3>
  <div
    className="container"
  >
    <form
      className="w-50 vh-100 d-flex flex-column mx-auto "
      id="contactForm"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="formName"
        >
          Name
        </label>
        <input
          className="form-control"
          defaultValue=""
          id="formName"
          name="name"
          onChange={[Function]}
          placeholder="Enter name"
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="formEmail"
        >
          Email address
        </label>
        <input
          className="form-control"
          defaultValue=""
          id="formEmail"
          name="email"
          onChange={[Function]}
          placeholder="Enter email"
          type="email"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="formMessage"
        >
          Message
        </label>
        <textarea
          className="form-control"
          defaultValue=""
          id="formMessage"
          name="message"
          onChange={[Function]}
          placeholder="Enter message"
          rows={10}
          style={
            Object {
              "resize": "none",
            }
          }
          type="text"
        />
      </div>
      <div
        className="d-flex justify-content-end"
      >
        <button
          className="btn btn-success"
          disabled={false}
          onSubmit={[Function]}
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</div>
`;
